narwhals/series.py

```diff
@@ -156,7 +159,9 @@ def __getitem__(self: Self, idx: int | slice | Sequence[int]) -> Any | Self:
             is_numpy_scalar(idx) and idx.dtype.kind in ("i", "u")
         ):
             return self._compliant_series[idx]
-        return self._from_compliant_series(self._compliant_series[idx])
+        return self._from_compliant_series(
+            self._compliant_series[to_native(idx, pass_through=True)]
```

Comment:
Kind of unrelated, but we have a method called `_extract_native` which actually maybe extracts compliant 🙈

Reply:
that's an awfully good point